src/core/ext/transport/chttp2/client/chttp2_connector.cc

```diff
@@ -171,13 +174,16 @@ void Chttp2Connector::OnHandshakeDone(void* arg, grpc_error_handle error) {
       self->Ref().release();  // Ref held by OnReceiveSettings()
       GRPC_CLOSURE_INIT(&self->on_receive_settings_, OnReceiveSettings, self,
                         grpc_schedule_on_exec_ctx);
-      self->Ref().release();  // Ref held by OnTimeout()
       grpc_chttp2_transport_start_reading(self->result_->transport,
                                           args->read_buffer,
                                           &self->on_receive_settings_, nullptr);
-      GRPC_CLOSURE_INIT(&self->on_timeout_, OnTimeout, self,
-                        grpc_schedule_on_exec_ctx);
-      grpc_timer_init(&self->timer_, self->args_.deadline, &self->on_timeout_);
+      self->timer_handle_ =
+          self->args_.channel_args.GetObject<EventEngine>()->RunAfter(
+              self->args_.deadline - Timestamp::Now(), [self = self->Ref()] {
+                ApplicationCallbackExecCtx callback_exec_ctx;
+                ExecCtx exec_ctx;
+                static_cast<Chttp2Connector*>(self.get())->OnTimeout();
+              });
```

Comment:
An alternative is to write something like:

```
// self->Ref() returns a RefCountedPtr<SubchannelConnector>, an
// implicit downcast will be used here:
RefCountedPtr<Chttp2Connector> cc = self->Ref();
self->timer_handle_ =
    self->args_.channel_args.GetObject<EventEngine>()->RunAfter(
        self->args_.deadline - Timestamp::Now(), [self = std::move(cc)] {
          ApplicationCallbackExecCtx callback_exec_ctx;
          ExecCtx exec_ctx;
          self->OnTimeout();
        });
```

I'm not sure which one is more readable or better in other ways. So went with the one without creating an extra `cc` variable.

Reply:
I like this form

Reply:
Done